modules/mutect2/1.0/mutect2.smk

```diff
@@ -95,18 +107,100 @@ rule _mutect2_run:
         op.as_one_line("""
         gatk Mutect2 {params.opts} -I {input.tumour_bam} -I {input.normal_bam}
         -R {input.fasta} -normal $(cat {input.normal_sm}) -O {output.vcf}
-        --germline-resource {input.gnomad} > {log.stdout} 2> {log.stderr}
+        --germline-resource {input.gnomad} -L {wildcards.chrom} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+def _mutect2_get_chr_vcfs(wildcards):
+    CFG = config["lcr-modules"]["mutect2"]
+    chrs = checkpoints._mutect2_input_chrs.get(**wildcards).output.chrs
+    with open(chrs) as file:
+        chrs = file.read().rstrip("\n").split("\n")
+    vcfs = expand(
+        CFG["dirs"]["mutect2"] + "{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}/chromosomes/{chrom}.output.vcf.gz",
+        chrom = chrs
+    )
+    return(vcfs)
+
+
+def _mutect2_get_chr_tbis(wildcards):
+    CFG = config["lcr-modules"]["mutect2"]
+    chrs = checkpoints._mutect2_input_chrs.get(**wildcards).output.chrs
+    with open(chrs) as file:
+        chrs = file.read().rstrip("\n").split("\n")
+    tbis = expand(
+        CFG["dirs"]["mutect2"] + "{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}/chromosomes/{chrom}.output.vcf.gz.tbi",
+        chrom = chrs
+    )
+    return(tbis)
+
+
+# Merge chromosome mutect2 VCFs from the same sample
+rule _mutect2_merge_vcfs:
+    input:
+        vcf = _mutect2_get_chr_vcfs,
```

Comment:
To avoid code duplication, couldn't you forego the redundant function to retrieve tbi files and instead just modify each string in the list returned by the first function using a lambda function that concatenates ".tbi". In fact, why are we even listing the tbi as an input for this?

Reply:
I was observing that when using scratch subdirectories the indices were not appearing alongside the VCF files unless they were inputs in the merge step, and the indices are needed for bcftools concat to work. I tried outputting both VCF and .tbi files from the same input function as a dictionary and using `unpack()`, but I was getting an error stating that the `vcf` key did not exist.

Reply:
I went back to see if it was possible to use a single input function and it seems to work now. I probably used a different key name than `vcf`.